chains/evm/voter/proposal.go

```diff
@@ -93,12 +94,24 @@ func (p *Proposal) Execute(client ChainClient) error {
 		return err
 	}
 	client.LockOpts()
+	defer client.UnlockOpts()
+
 	opts, err := client.UnsafeOpts()
 	if err != nil {
 		return err
 	}
 	opts.GasLimit = uint64(2000000)
-	tx := evmtransaction.NewTransaction(opts, p.BridgeAddress, big.NewInt(0), cId, input)
+
+	var tx evmclient.CommonTransaction
+	if activated, err := client.IsEIP1559Activated(); err != nil {
+		if activated {
+			tx = evmtransaction.NewDynamicFeeTransaction(cId, opts.Nonce.Uint64(), &p.BridgeAddress, big.NewInt(0), opts.GasTipCap, opts.GasFeeCap, opts.GasLimit, input)
+		} else {
+			tx = evmtransaction.NewTransaction(opts.Nonce.Uint64(), &p.BridgeAddress, big.NewInt(0), opts.GasLimit, opts.GasPrice, input)
+		}
+	} else {
```

Comment:
Is this also meant to be within the error block?

Reply:
Same for these